packages/block-editor/src/components/writing-flow/index.js

```diff
@@ -99,6 +110,12 @@ class WritingFlow extends Component {
 		this.verticalRect = null;
 	}
 
+	switchToEditMode() {
+		if ( this.props.keyboardMode !== 'edit' ) {
```

Comment:
yeah, we probably save some milliseconds this way but probably not that much. Also, there's no need to do it in the reducer since it's just a scalar value and it will still be considered as the same value.